components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Storage/PublishedNodesJobConverter.cs

```diff
@@ -259,6 +259,10 @@ public class PublishedNodesJobConverter {
         /// </summary>
         private static string GetUniqueWriterId(IEnumerable<PublishedDataSetSourceModel> set, PublishedDataSetSourceModel model) {
             var result = model.Connection.Id;
+            if (string.IsNullOrEmpty(model.Connection.Id)) {
```

Comment:
I see the bug here. The solution though is is not the right one.
it should be something like:
```
            var result = model.Connection.Id;
            var subset = set.Where(x => x.Connection.Id == model.Connection.Id).ToList();
            if (subset.Count > 1) {
                result += !string.IsNullOrEmpty(result) ? "_" : string.Empty;
                result += $"{model.SubscriptionSettings.PublishingInterval.GetValueOrDefault().TotalMilliseconds}";
                if (subset.Where(x => x.SubscriptionSettings.PublishingInterval == model.SubscriptionSettings.PublishingInterval).Count() > 1) {
                    result += $"_{model.PublishedVariables.PublishedData.First().PublishedVariableNodeId}";
                }
            }
            else {
                result ??= model.SubscriptionSettings.PublishingInterval.GetValueOrDefault().TotalMilliseconds.ToString();
            }
            return result;
```

Reply:
fixed